core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -122,7 +126,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
           context.reply(RegisteredExecutor)
-
+          latestTokens.foreach(x => context.reply(NewTokens(x)))
```

Comment:
Isn't that the case with any of the messages being sent? I have no idea how akka handles message delivery failure. 

I can try again, but there were so many test failures it was painful to fix.